community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/input/ByteBufferReadableChannel.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.unsafe.impl.batchimport.input;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import org.neo4j.kernel.impl.transaction.log.ReadableClosableChannel;
+
+/**
+ * Wraps a byte[] -> {@link ByteBuffer} -> {@link ReadableClosableChannel}?
+ */
+public class ByteBufferReadableChannel implements ReadableClosableChannel
```

Comment:
Judging by usages, it seems that ValueType is working (by proxy) on ByteBuffers anyway, rather than real channels. Maybe that should be surfaced in the parameter lists to the read and write methods on ValueType? Then you wouldn't need these classes, but you'd have to fix that (as far as I have found) one place where a "real" channel is being worked on.

Reply:
I know I struggled a bit with that back and forth... I'd be happy to let it rest, but the wrapping is a bit annoying, even though it's just a very peripheral thing in the outer corner of the import code.